sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp

```diff
@@ -518,74 +633,104 @@ bool test_fp_types(queue q, const Config &cfg) {
                                  ImplFcmpwr<sycl::half, N>>) {
       auto dev = q.get_device();
       if (dev.has(sycl::aspect::fp16)) {
-        passed &= test<sycl::half, N, Op, UseMask, UsePVCFeatures>(q, cfg);
+        passed &= run_test<UseAcc, sycl::half, N, Op, UseMask, UsePVCFeatures>(
+            q, cfg);
       }
     }
   }
-  passed &= test<float, N, Op, UseMask, UsePVCFeatures>(q, cfg);
+  passed &= run_test<UseAcc, float, N, Op, UseMask, UsePVCFeatures>(q, cfg);
 #ifndef CMPXCHG_TEST
-  if (q.get_device().has(sycl::aspect::atomic64) &&
-      q.get_device().has(sycl::aspect::fp64)) {
-    passed &= test<double, N, Op, UseMask, UsePVCFeatures>(q, cfg);
+#ifdef __ESIMD_FORCE_STATELESS_MEM
+  constexpr bool test_double = true;
+#else
+  constexpr bool test_double = !UseAcc;
```

Comment:
Why double is not tested in acc+stateful mode?

Reply:
for that intrinsic, we only support 4 byte data, and assert that. see [here](https://github.com/intel/llvm/blob/156b730a6886b1394b7b91e2455d6e6b42ac6ec1/sycl/include/sycl/ext/intel/esimd/memory.hpp#L5131)

Reply:
double should be supported in LSC mode. Also double as FP type is supported only in LSC mode and in both(acc and usm) variants, right?